src/reporter/interfaces.ts

```diff
@@ -1,6 +1,7 @@
 import { Writable as WritableStream } from 'stream';
 
 export interface ReporterPlugin {
+    reportInit(): void;
```

Comment:
There is a difference between declaration and implementation of the `reporterInit` method.

Declaration - `reportInit(): Promise<void>`
Implementation - `reportInit(): Promise<{ error: Error }>`

I guess we can raise error in the `reporterInit` and handle this error later in code instead of adding the return value of reporter methods.

Reply:
ok